Heimdallr/Core/Heimdallr.swift

```diff
@@ -155,10 +155,9 @@ public let HeimdallrErrorNotAuthorized = 2
                     completion(.Failure(error))
                 }
             } else {
-                switch OAuthError.decode(data!) {
-                case let .Success(error):
+                if let error = OAuthError.decode(data: data!) {
```

Comment:
Would you mind manually unwrapping `data`, potentially failing by not satisfying a condition or using `fatalError` with an appropriate error message if `data` is `nil`?